selfie.c

```diff
@@ -8819,18 +9381,26 @@ uint64_t handle_timer(uint64_t* context) {
   set_exception(context, EXCEPTION_NOEXCEPTION);
 
   if (symbolic) {
-    print("(push 1)\n");
+    print(";(push 1)\n");
```

Comment:
Yes, that is not necessary. I removed that.

Reply:
But where do you push? Even if you always merge you still get more than one formula to solve, right?

Reply:
I push only when a path does properly exit, if a path is reaches its maximal execution depth (i.e. is timed out), I do not push since this path did not properly finish execution.
Normally all paths are merged, this means in particular that the last if statement in the examples is also merged and there is overall only one formula to solve.

Reply:
But what if there is an error on a path where you reach the maximal execution depth? Say there is a division-by-zero error on such a path and no other path. In that case, you'd miss that, unnecessarily, no?

Reply:
The division-by-zero error would cause the path to stop executing and would result in an exit (the method handle_division_by_zero returns EXIT), so this path would exit 'properly' and would be pushed. So nothing should be missed, or am I looking at this the wrong way?

Reply:
My previous answer is not correct. I can see what you mean now. I will have to take a look at it, I will let you know.

Reply:
Division by zero is only handled during concrete execution on mipster level. Your goal is to encode binaries in such a way that, using an SMT solver, you can compute input that will make the binaries do a division by zero during concrete execution. The same is true for other errors such as unsafe memory access.

Reply:
So, not pushing timed-out paths misses such errors if they occur on parts of the timed-out paths that are not merged into other paths for which formulae are generated.

Reply:
That is true, I will work on that.

Reply:
Done (for division by zero and invalid concrete memory addresses). However, I noticed that during symbolic execution it is only checked whether the virtual address is a valid address, not if the virtual address is also mapped, which could be the case in `constrain_ld`. I did not modify that since I do not know if there is a specific reason for that. Should I implement that check?

Reply:
The reason is that the property of a page to be mapped is complex. It depends on the kernel that does the mapping of pages. So, no, don’t implement that check.